### ISSUE TITLE
How to install Bower component from Stash repo with JSPM

### ISSUE THREAD

**Issue by jansensan** [Orbs/jspm-git#25]
This is a question [I asked on StackOverflow](http://stackoverflow.com/questions/34734823/how-to-install-bower-component-from-stash-repo-with-jspm) already, however you may know about this as well.

My client uses Stash to host project repositories. We use a private Bower registry for our private libraries.

In this new project I am building, I am using JSPM. I have used jspm-git to create a stash registry. Getting the public libraries such as jQuery, Angular and Lodash obviously works fine.

I get into an issue when I run this:

    jspm install mapped-name=stash:project-name/lib-name

Obviously the actual lib names are redacted to respect my client's intellectual property.

This is the error I get:

    warn Error on build
     TypeError: undefined is not a function
         at Object.posix.isAbsolute (path.js:479:15)
         at Object.posix.normalize (path.js:461:26)
         at GitLocation.build ([REDACTED]/node_modules/jspm-git/git.js:426:17)
         at [REDACTED]/node_modules/jspm/lib/registry.js:117:30
         at lib$rsvp$$internal$$tryCatch ([REDACTED]/node_modules/jspm/node_modules/rsvp/dist/rsvp.js:493:16)
         at lib$rsvp$$internal$$invokeCallback ([REDACTED]/node_modules/jspm/node_modules/rsvp/dist/rsvp.js:505:17)
         at [REDACTED]/node_modules/jspm/node_modules/rsvp/dist/rsvp.js:1001:13
         at lib$rsvp$asap$$flush ([REDACTED]/node_modules/jspm/node_modules/rsvp/dist/rsvp.js:1198:9)
         at process._tickCallback (node.js:355:11)

I do not know if I forgot a step, or if I misconfigured something, or if there is even a bug with JSPM or jspm-git. Any guidances to help me debug this would be appreciated.

**Comment by subesokun** [comment]
Hi @jansensan! Something seems to be wrong with the "main" property of your bower.json as the error gets thrown here https://github.com/Orbs/jspm-git/blob/master/git.js#L426
How does your bower.json looks like? But indeed the error handling should be improved here to avoid such errors and give a reasonable feedback what went wrong.

**Comment by jansensan** [comment]
After careful exploration of libraries that worked properly, it turned out to be a missing [`publishConfig`](https://docs.npmjs.com/files/package.json#publishconfig) entry in `package.json`. I added it and it fixed it.

However, I would suggest a more detailed error message, if at all possible. Simply saying "Error on build" leads to misinterpretations: why would JSPM or jspm-git attempt to build? If there is any way to inform of  what would be missing from the `package.json`, it would help greatly.

Update: I just saw your reply as I am writing this! :)
